compute/baremetal.py

```diff
@@ -74,15 +118,376 @@ def __init__(self, **params) -> None:
+        LOG.info(
+            "Re-image successfull. The cuurent os-version of {node} "
+            "is {os}".format(node=self.node, os=os_version)
+        )
+
+    def _unlock_and_lock_node(self):
+        """
+        Fetch the current lock status of the nodes. Unlock if from the
+        previous user, lock it under the teuthology creds.
+        :return: True/False
+        """
+        t_user = "{}@{}".format(
+            self.teuthology_creds.get("username"), self.teuthology_creds.get("hostname")
+        )
+
+        # Get the lock status of the node
+        lock_status = self._get_node_lock_status()
+
+        # Check 1: Verify the node is already locked, if yes, proceed to unlock, and finally lock
+        if lock_status["is_locked"]:
+            locked_user = lock_status["locked_user"]
+            self.node_owner = locked_user
+            # Now that the machine is locked, check if its already locked by the teuthology user
+            if locked_user != t_user:
+                # If not, then perform unlock
+                LOG.debug("{} is currently locked by {}".format(self.node, locked_user))
+
+                # Unlock the node using the current locked user data
+                cmd = "teuthology-lock --unlock --owner {} {}".format(
+                    locked_user, self.node
+                )
+                _, out, _ = self._execute_command_in_venv(cmd)
+                print(out.read().decode(encoding="UTF-8"))
+            else:
+                LOG.debug(
+                    "The node is already locked by teuthology user {}."
+                    "Skipping the locking steps".format(t_user)
+                )
+                return
+
+        # Now that the machine is unlocked, perform lock using t_user
+        cmd = "teuthology-lock --lock --owner {} {}".format(t_user, self.node)
+        _, out, _ = self._execute_command_in_venv(cmd)
+        print(out.read().decode(encoding="UTF-8"))
+
+        # Verify the lock command has been successful
+        lock_status = self._get_node_lock_status()
+        if not (lock_status["is_locked"] and lock_status["locked_user"] == t_user):
+            LOG.error(
+                "Lock failed. {} still under {}".format(
+                    self.node, lock_status["locked_user"]
+                )
+            )
+            exit(1)
+
+        # The lock command executed successful
+        LOG.debug("{} successfully locked by {}".format(self.node, t_user))
+
+    def _relock_node(self):
+        """
+        Relock the node to the actual owner. This step is done to ensure that the node is assigned back to
+        user and not reserved under the teuthology master node.
+        :return: (bool) based on lock status
+        """
+        # Unlock the node from teuthology master node
+        # Unlock the node using the current locked user data
+        locked_user = "{}@{}".format(
+            self.teuthology_creds.get("username"), self.teuthology_creds.get("hostname")
+        )
```

Comment:
can we use f-string instead of format
locked_user = f"{self.teuthology_creds.get("username")}@{self.teuthology_creds.get("hostname")}

Reply:
Noted. Will change the .format strings to f-string